### PR TITLE
#809 update prettier eslint

### LINKED ISSUE #809: Update prettier, eslint formatting
Build Number: 2.2.0-rc0

Description: Placeholder issue for making changes to prettier and eslint config based on feedback after PR #797.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #809.

### CODE REVIEW

File: src/database/DataTypes/Item.js [hunk 1]
```diff
@@ -50,17 +50,15 @@ export class Item extends Realm.Object {
...
   get earliestExpiringBatch() {
-    const { batches } = this;
+    // If no batches associated with this item, return null.
+    if (this.batches.length === 0) return null;
 
-    if (batches.length === 0) return null;
-    let earliestBatch = batches.find(batch => {
-      return batch.totalQuantity > 0;
-    });
+    let earliestBatch = this.batches.find(batch => batch.totalQuantity > 0);
```
Comment (andreievg) [inline src/database/DataTypes/Item.js]: why not ` const { batches } = this ? @wlthomson

File: src/database/DataTypes/RequisitionItem.js [hunk 2]
```diff
@@ -30,8 +30,9 @@ export class RequisitionItem extends Realm.Object {
   }
 
   get suggestedQuantity() {
-    const daysToSupply = this.requisition ? this.requisition.daysToSupply : 0;
-    return Math.ceil(Math.max(this.dailyUsage * daysToSupply - this.stockOnHand, 0));
+    const { daysToSupply } = this.requisition || { daysToSupply: 0 };
```
Comment (andreievg) [inline src/database/DataTypes/RequisitionItem.js]: This doesn't look like eslint change. (pretty smart though, but previous code is more readable)

File: src/database/DataTypes/RequisitionItem.js [hunk 3]
```diff
@@ -43,19 +44,21 @@ export class RequisitionItem extends Realm.Object {
...
+    return totalQuantity;
   }
 
   setSuppliedQuantity(database, newValue) {
     if (this.requisition.isFinalised || this.requisition.isRequest) {
       throw new Error('Cannot set supplied quantity for Finalised or Request Requisition');
     }
 
-    const transactionItem = this.linkedTransactionItem;
+    const { linkedTransactionItem: transactionItem } = this;
```
Comment (andreievg) [inline src/database/DataTypes/RequisitionItem.js]: As per our chart about destructuring this (I am not against it btw).

I can see you are reverting some of the destructuring of this, but the change here is unnecessary imo

File: src/database/DataTypes/RequisitionItem.js [hunk 4]
```diff
@@ -43,19 +44,21 @@ export class RequisitionItem extends Realm.Object {
     const { availableQuantity } = this.linkedTransactionItem;
     const { totalQuantity } = this.item;
 
-    return this.linkedTransactionItem ? availableQuantity : totalQuantity;
+    // TODO: this.linkedTransactionItem and this.item can both be falsey, unhandled possible return
+    if (availableQuantity) return availableQuantity;
```
Comment (andreievg) [inline src/database/DataTypes/RequisitionItem.js]: no need to refactor code in this PR, just applying eslint/prettier (can leave the comment though)

File: src/database/DataTypes/StocktakeBatch.js [hunk 5]
```diff
@@ -20,7 +20,8 @@ export class StocktakeBatch extends Realm.Object {
    * @return {boolean} True if |stockOnHand + stocktakebatch.difference| is negative.
    */
   get isReducedBelowMinimum() {
-    const stockOnHand = this.itemBatch.totalQuantity;
+    const { itemBatch } = this;
+    const { totalQuantity: stockOnHand } = itemBatch;
```
Comment (andreievg) [inline src/database/DataTypes/StocktakeBatch.js]: I would revert this back to original (again as per this destructuring)

**Comment by wlthomson** [comment]
> reviewed up to createRecord.
> 
> Not sure why extra code refactoring are in this PR.
> Also not sure the repercussion of refactoring from function to const (especially on exports, could end up having circular dependency and compilation errors)

Regarding converting function declarations to arrow functions, my motivation came out of https://eslint.org/docs/rules/no-use-before-define. Function declarations are hoisted to file scope, which means you can use functions before defining them. Eslint doesnt like this and IMO it's jarring to read references to a function that hasn't been imported or defined (but is defined later in the same file).

File: src/database/DataTypes/StocktakeBatch.js (hunk 5 above)
Comment (wlthomson) [reply src/database/DataTypes/StocktakeBatch.js]: Reverted.

File: src/database/DataTypes/RequisitionItem.js (hunk 4 above)
Comment (wlthomson) [reply src/database/DataTypes/RequisitionItem.js]: Reverted.

File: src/database/DataTypes/RequisitionItem.js (hunk 3 above)
Comment (wlthomson) [reply src/database/DataTypes/RequisitionItem.js]: Reverted.

File: src/database/DataTypes/RequisitionItem.js (hunk 2 above)
Comment (wlthomson) [reply src/database/DataTypes/RequisitionItem.js]: Reverted.

File: src/database/DataTypes/RequisitionItem.js (hunk 4 above)
Comment (wlthomson) [reply src/database/DataTypes/RequisitionItem.js]: Reverted.

**Pull request merged**